travel-planner/travel-planner-ui/src/Components/SelectDirections/SelectDirections.jsx

```diff
@@ -0,0 +1,116 @@
+import "./SelectDirections.css";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import Dropdown from "react-bootstrap/Dropdown";
+import Form from "react-bootstrap/Form";
+
+export default function SelectDirections({ PORT, mapId }) {
+	// get all directions from backend
+	const [directions, setDirections] = useState(null);
+  const [curDirections, setCurDirections] = useState(null);
+
+
+	const CustomToggle = React.forwardRef(({ children, onClick }, ref) => (
+		<a
+			href=""
+			ref={ref}
+			onClick={(e) => {
+				e.preventDefault();
+				onClick(e);
+			}}
+		>
+			{children}
+			&#x25bc;
+		</a>
+	));
```

Comment:
I believe you can extract this to a component, if you feel like it will only be used on this file it is okay to extract to a different top level function on this same file